src/components/PasswordToggle.vue

```diff
@@ -0,0 +1,28 @@
+<script setup>
+import { ref } from "vue";
+
+let showPassword = ref(false);
+</script>
+
+<template>
+  <span class="toggle_button">
+    <i class="bi toggle_icon" :class="showPassword ? 'bi-eye' : 'bi-eye-slash'" id="showPassword"
+      @click="showPassword = !showPassword"></i>
```

Comment:
When the icon is clicked, emit an event which will trigger the parent component to change the state.